core/storage/skill/gae_models.py

```diff
@@ -154,3 +154,35 @@ class SkillSummaryModel(base_models.BaseModel):
     # model was created).
     skill_model_created_on = ndb.DateTimeProperty(required=True, indexed=True)
     version = ndb.IntegerProperty(required=True)
+
+
+class SkillRightsSnapshotMetadataModel(base_models.BaseSnapshotMetadataModel):
+    """Storage model for the metadata for a skill rights snapshot."""
+    pass
+
+
+class SkillRightsSnapshotContentModel(base_models.BaseSnapshotContentModel):
+    """Storage model for the content of a skill rights snapshot."""
+    pass
+
+
+class SkillRightsModel(base_models.VersionedModel):
+    """Storage model for the rights related to a skill.
+
+    The id of each instance is the id of the corresponding skill.
+    """
+    SNAPSHOT_METADATA_CLASS = SkillRightsSnapshotMetadataModel
+    SNAPSHOT_CONTENT_CLASS = SkillRightsSnapshotContentModel
+    ALLOW_REVERT = False
+
+    # The user_id of the creator this skill.
+    creator_id = ndb.StringProperty(indexed=True)
+    # Whether the skill is private.
+    skill_is_private = ndb.BooleanProperty(
+        indexed=True, required=True, default=True)
+
+    @classmethod
+    def get_unpublished_skills_by_creator_id(cls, user_id):
+        return cls.query(
+            cls.creator_id == user_id,
+            cls.skill_is_private is True)
```

Comment:
I think this is supposed to be "==", not "is". You might need a pylint pragma here if that causes errors.

Reply:
done